suite/tests/client-interface/drwrap-drreg-test.dll.c

```diff
@@ -80,7 +80,12 @@ module_load_event(void *drcontext, const module_data_t *mod, bool loaded)
 
         addr_two_args = (app_pc)dr_get_proc_address(mod->handle, "two_args");
         CHECK(addr_two_args != NULL, "cannot find lib export");
+#if defined(RISCV64)
+        bool ok = drwrap_wrap_ex(addr_two_args, wrap_pre, wrap_post, 0,
+                                 DRWRAP_CALLCONV_RISCV_LP64 | DRWRAP_REPLACE_RETADDR);
```

Comment:
DRWRAP_REPLACE_RETADDR has some risks and should not be needed.  (As its docs say, it requires stack scanning "on AArchXX": please update that doc and the code to be sure it's doing the same on RISC-V.). It sounds like something is wrong if things don't work without this.  If it's left here there should be a TODO comment explaining that we need to get the defaults working for which investigation is needed.  If the defaults don't work then every user on RISC-V is going to hit these same failures you're hitting when using drwrap.

Reply:
I removed DRWRAP_REPLACE_RETADDR and tested (on my client and DynamoRIO tests): it seems to work for post-wrappers. But with the stack scanning things: it is implemented in get_cur_xsp which is now under the ARCHXX guards and used for drwrap_replace_native_fini() function. It is validated in separate test which is not implemented currently for RISCV, so I prefer to not touching that code in this PR.

Reply:
Please document publicy in the DRWRAP_REPLACE_RETADDR doxygen doc that RISC-V is not supported for that flag, with a non-public `TODO i#3544`.

Reply:
Done